lib/chunky_png/palette.rb

```diff
@@ -21,8 +21,10 @@ class Palette < SortedSet
     #   which they appeared in the palette chunk, so that this array can be
     #   used for decoding.
     def initialize(enum, decoding_map = nil)
-      super(enum)
+      super(enum.sort.freeze)
       @decoding_map = decoding_map if decoding_map
+      @encoding_map = {}
+      freeze
```

Comment:
I freeze the palette to validate my assumption that elements are not added to the set.